mmv1/third_party/terraform/tests/resource_gkeonprem_bare_metal_cluster_test.go.erb

```diff
@@ -0,0 +1,133 @@
+<% autogen_exception -%>
+package google
+<% if version == 'private' -%>
+
+import (
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+)
+
+func TestAccgkeonpremBareMetalCluster_update(t *testing.T) {
+	t.Parallel()
+
+	context := map[string]interface{}{
+		"random_suffix": randString(t, 10),
+	}
+
+	vcrTest(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProvidersOiCS,
+		CheckDestroy: testAccCheckgkeonpremBareMetalClusterDestroyProducer(t),
+		Steps: []resource.TestStep{
+			{
+				Config: testAccgkeonpremBareMetalCluster_gkeonpremBareMetalClusterExample(context),
+			},
+			{
+				ResourceName:            "google_gkeonprem_bare_metal_cluster.cluster1",
+				ImportState:             true,
+				ImportStateVerify:       true,
+				ImportStateVerifyIgnore: []string{"name", "location"},
+			},
+			{
+				Config: testAccgkeonpremBareMetalCluster_update(context),
+			},
+			{
+				ResourceName:            "google_gkeonprem_bare_metal_cluster.cluster1",
+				ImportState:             true,
+				ImportStateVerify:       true,
+				ImportStateVerifyIgnore: []string{"name", "location"},
+			},
+		},
+	})
+}
+
+func testAccgkeonpremBareMetalCluster_update(context map[string]interface{}) string {
+	return Nprintf(`
+
+resource "google_gkeonprem_bare_metal_cluster" "cluster1" {
+	name = "cluster-basic%{random_suffix}"
+	location = "us-west1"
+	admin_cluster_membership = "projects/624768296925/locations/global/memberships/td7bb198b0744eeb-cluster"
```

Comment:
This doesn't feel like a good thing to have in a test. Can we set up the admin cluster as part of this test?

Reply:
Unfortunately no. We're releasing our products in reverse order first the user clusters then the admin clusters. So, for now, there is no direct API or TF support for admin clusters. We can update this in the future.

For testing purposes, this is just a fake input, but until

Reply:
Then it doesn't really make sense to commit this test until that setup is possible, right?